### PR TITLE
feat: set  metadata for launched consumer chains

### PR DESCRIPTION
**Pull request by mpoke**
## Description

Closes: #XXXX

Also deal with TODOs in the v20 upgrade handler. 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [ ] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] Added `!` to the type prefix if API, client, or state breaking change (i.e., requires minor or major version bump)
* [ ] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
* [ ] Provided a link to the relevant issue or specification
* [ ] Followed the guidelines for [building SDK modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules)
* [ ] Included the necessary unit and integration [tests](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#testing)
* [ ] Added a changelog entry in `.changelog` (for details, see [contributing guidelines](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#changelog))
* [ ] Included comments for [documenting Go code](https://blog.golang.org/godoc)
* [ ] Updated the relevant documentation or specification
* [ ] Reviewed "Files changed" and left comments if necessary 
* [ ] Confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed `!` in the type prefix if API or client breaking change
* [ ] confirmed all author checklist items have been addressed 
* [ ] reviewed state machine logic
* [ ] reviewed API design and naming
* [ ] reviewed documentation is accurate
* [ ] reviewed tests and test coverage

### CODE REVIEW

File: app/upgrades/v20/upgrades.go [hunk 1]
```diff
@@ -345,19 +396,18 @@ func MigrateConsumerAdditionProposal(
...
 		Metadata:    "TBA",
 	}
 	err := providertypes.ValidateConsumerMetadata(metadata)
-	// TODO: avoid returning an error here;
-	// for this, we need to make sure the validation works always
-	// (e.g., truncate the fields)
-	return metadata, err
+	if err != nil {
+		metadata.Name = providertypes.TruncateString(metadata.Name, providertypes.MaxNameLength)
+		metadata.Description = providertypes.TruncateString(metadata.Description, providertypes.MaxDescriptionLength)
```
Comment (insumity) [inline app/upgrades/v20/upgrades.go:421]: In case of error, do we still want `Metadata` to be set to `"TBA"`?

File: app/upgrades/v20/upgrades.go [hunk 2]
```diff
@@ -645,3 +696,62 @@ func MigrateChangeRewardDenomsProposal(
 	}
 	return nil
 }
+
+// SetICSConsumerMetadata sets the metadata for launched consumer chains
+func SetICSConsumerMetadata(ctx sdk.Context, providerKeeper providerkeeper.Keeper) error {
+	for _, consumerID := range providerKeeper.GetAllActiveConsumerIds(ctx) {
```
Comment (insumity) [inline app/upgrades/v20/upgrades.go:715]: We have `GetAllLaunchedConsumerIds` so we can use that one directly (?)
s/consumerID/consumerId.

**Review by insumity — APPROVED** [review]

File: app/upgrades/v20/upgrades.go (hunk 1 above)
Comment (mpoke) [reply app/upgrades/v20/upgrades.go:421]: Yeah, why not? We'll need to do this in a subsequent upgrade as we don't know what consumers will send props in this period.

File: app/upgrades/v20/upgrades.go (hunk 2 above)
Comment (mpoke) [reply app/upgrades/v20/upgrades.go:715]: > s/consumerID/consumerId.

The linter doesn't like consumerId for some reason.

Comment (mpoke) [reply app/upgrades/v20/upgrades.go:715]: > We have GetAllLaunchedConsumerIds so we can use that one directly (?)

It doesn't really matter. Leave it like this. Before the v20 upgrade, the number of consumer IDs is low. :)

**Review by bermuell — APPROVED** [review]

**Review by sainoe — APPROVED** [review]
LGTM

**Pull request merged**